src/Facebook/GraphNodes/GraphEvent.php

```diff
@@ -0,0 +1,224 @@
+<?php
+/**
+ * Copyright 2014 Facebook, Inc.
+ *
+ * You are hereby granted a non-exclusive, worldwide, royalty-free license to
+ * use, copy, modify, and distribute this software in source code or binary
+ * form for use in connection with the web services and APIs provided by
+ * Facebook.
+ *
+ * As with any software that integrates with the Facebook platform, your use
+ * of this software is subject to the Facebook Developer Principles and
+ * Policies [http://developers.facebook.com/policy/]. This copyright notice
+ * shall be included in all copies or substantial portions of the software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ */
+namespace Facebook\GraphNodes;
+
+/**
+ * Class GraphEvent
+ *
+ * @package Facebook
+ */
+class GraphEvent extends GraphObject
+{
+    /**
+     * @var array Maps object key names to Graph object types.
+     */
+    protected static $graphObjectMap = [
+        'cover' => '\Facebook\GraphNodes\GraphCoverPhoto',
+        'place' => '\Facebook\GraphNodes\GraphPage',
+        'picture' => '\Facebook\GraphNodes\GraphPicture',
+        // @todo parent_group must be cast to GraphGroup once supported
+        // 'parent_group' => '\Facebook\GraphNodes\GraphGroup',
+    ];
+
+    /**
+     * Returns the `id` (The event ID) as string if present.
+     *
+     * @return string|null
+     */
+    public function getId()
+    {
+        return $this->getProperty('id');
+    }
+    /**
+     * Returns the `cover` (Cover picture) as GraphCoverPhoto if present.
+     *
+     * @return GraphCoverPhoto|null
+     */
+    public function getCover()
+    {
+        return $this->getProperty('cover');
+    }
+    /**
+     * Returns the `description` (Long-form description) as string if present.
+     *
+     * @return string|null
+     */
+    public function getDescription()
+    {
+        return $this->getProperty('description');
+    }
+    /**
+     * Returns the `end_time` (End time, if one has been set) as DateTime if present.
+     *
+     * @return DateTime|null
+     */
+    public function getEndTime()
+    {
+        return $this->getProperty('end_time');
+    }
+    /**
+     * Returns the `is_date_only` (Whether the event only has a date specified, but no time) as bool if present.
+     *
+     * @return bool|null
+     */
+    public function getIsDateOnly()
```

Comment:
> We can add right methods in 4.1 and mark old ones as deprecated, and in master (wich is not used yet) we can simply change the methods.

Love that idea. @gfosco Which naming schema do you prefer for all boolean getter methods:

`get{SnakeCaseOfFieldName}()` or `{snakeCaseOfFieldName}()`

For example the `is_date_only` field would be either:

`getIsDateOnly()` or `isDateOnly()`

- `get{FieldName}()` is consistent with naming schema of the other methods but doesn't read very well.
- `{fieldName}()` reads a lot better, but isn't consistent with the other getter methods.